cellcounter/cc_kapi/models.py

```diff
@@ -40,29 +62,27 @@ def set_keymaps(self, new_mapping_list):
         """new_mapping_list is a list of KeyMap objects"""
         self._sync_keymaps(new_mapping_list)
 
-    def set_primary(self):
-        self._sync_primary_flags()
-
-    def delete(self):
-        """Custom delete method, covering setting primary flags"""
-        user = self.user
-        primary = self.is_primary
-
-        super(Keyboard, self).delete()
-        if primary:
-            keyboards = user.keyboard_set.all()
-            if keyboards:
-                keyboards[0].set_primary()
-
     def save(self, force_insert=False, force_update=False, using=None,
              update_fields=None):
+
+        if self.user:
+            self.last_modified = timezone.now()
+
         super(Keyboard, self).save(force_insert=False, force_update=False, using=None,
                                    update_fields=None)
-        if self.is_primary:
-            self.set_primary()
-
 
 class KeyMap(models.Model):
     cellid = models.ForeignKey(CellType)
     key = models.CharField(max_length=1)
-    keyboards = models.ManyToManyField(Keyboard, related_name='mappings')
\ No newline at end of file
+    keyboards = models.ManyToManyField(Keyboard, related_name='mappings')
+
+class DefaultKeyboards(models.Model):
```

Comment:
In the interests of using singular nouns for models, perhaps consider 'KeyboardProfile' or 'UserKeyboardProfile' as an alternative?